### ISSUE TITLE
Cypress is stopped and json error is displayed on console

### ISSUE THREAD

**Issue by arun290636** [cypress-io/cypress#1756]
Test execution is stopped.

### Current behavior: 

Below error is displayed on command prompt.
```
{ Error: unexpected end of file
    at Gunzip.zlibOnError (zlib.js:152:15)
 errno: -5, code: 'Z_BUF_ERROR' }
Error: unexpected end of file
    at Gunzip.zlibOnError (zlib.js:152:15)
```

cypress log:
![image](https://user-images.githubusercontent.com/1212793/40361542-98c259ba-5dc1-11e8-9de6-69d994556351.png)

### Desired behavior:

Test should work.
### Steps to reproduce:

Testing internal application

### Versions  
Latest

**Comment by jamesmckenzie** [comment]
we're experiencing the same issue with some urls in our tests. I've tried with the example above and was able to reproduce the error by disabling **chromeWebSecurity**

**Comment by neutcomp** [comment]
Just to inform you that we have the same issue at our end. It crashes randomly at any given moment. Also windows 7, Cypress 3.0.1 and latest chrome 67.0.3396.62. 

@jamesmckenzie does it solve if we enable chomeWebSecurity? Or does cypress then do not run anymore?

**Comment by jamesmckenzie** [comment]
@neutcomp to clarify, disabling chromeWebSecurity doesn't fix the issue, it just allows the above example to work without failing due to the same-origin-policy restrictions.

**Comment by arun290636** [comment]
@jamesmckenzie @neutcomp @ValerieThoma @jennifer-shehane 
It will be really helpful if we can get below issue fixed:
Below error is displayed on command prompt.
{ Error: unexpected end of file
at Gunzip.zlibOnError (zlib.js:152:15)
errno: -5, code: 'Z_BUF_ERROR' }
Error: unexpected end of file
at Gunzip.zlibOnError (zlib.js:152:15)

**Comment by arun290636** [comment]
@jennifer-shehane @ValerieThoma @brian-mann Guyes can you please run blow test few times:

OS- Windows 7
NPM version 5.6
Cypress version 3.1.0

Add   "chromeWebSecurity" : false in cypress.json

Add Cypress.on('uncaught:exception', (err, runnable) => {
    // returning false here prevents Cypress from
    // failing the test
    return false
})#

in Command.js

Code:

describe('checkout', function () {
    it('add card details', function () {

      cy.visit('http://www.smartbox.com/fr/')
      window.top.Cypress
   cy.get('button.button.primary.js-accept-cookies.cookie-consent__continue.gdpr-modal-button.gdpr-modal-button-continue')
   .click()
   
      cy.get('input.search-input').first()
           .type('hotels')
       cy.get('span.si.si-search').first()
           .click()
               cy.get('a.thematic__wrapper-link').first()
           .click()
       cy.get('span.add-to-cart__content').first()
           .click()
       cy.get('a.button.primary.no-margin-bottom')
           .click()
           cy.get('span.si.si-cart').last()
           .click({ force: true })
           cy.get('input#checkout-email-address')
    .type('abc@gmail.com')
           cy.get('input#shipping_lastname')
         .type('lname')
       cy.get('input#shipping_firstname')
         .type('fname')
       cy.get('input#shipping_street_1')
         .type('stret1')
       cy.get('input#shipping_street_2')
         .type('stret2')
       cy.get('input#shipping_postcode')
         .type('12345')
       cy.get('input#shipping_city')
         .type('paris')
       cy.get('input#shipping_telephone')
       .clear()
       .type('0123456789')
       cy.get('button#to-payment.button.primary.large-5.medium-5.small-12.no-margin')
         .click()
       cy.get('input#card-number').clear()
       .type('4111111111111111')
       cy.get('input#card-owner')
       .type('Test')
       cy.get('.month.expand.month').select('1')
       cy.get('.year.expand').select('2020')
       cy.get('input#card-security-code')
       .type('123')
})
})

Cypress is stopped with below error:
{ Error: unexpected end of file
    at Gunzip.zlibOnError (zlib.js:152:15)
 errno: -5, code: 'Z_BUF_ERROR' }
Error: unexpected end of file
    at Gunzip.zlibOnError (zlib.js:152:15)

**Comment by jennifer-shehane** [comment]
@arun290636 When I run the code above, I get the following error where it clicks on the 'span.si.si-cart'

```
CypressError: Cypress detected a cross origin error happened on page load:

  > Blocked a frame with origin "http://www.smartbox.com" from accessing a cross-origin frame.

Before the page load, you were bound to the origin policy:
  > http://smartbox.com

A cross origin error happens when your application navigates to a new superdomain which does not match the origin policy above.

This typically happens in one of three ways:

1. You clicked an <a> that routed you outside of your application
2. You submitted a form and your server redirected you outside of your application
3. You used a javascript redirect to a page outside of your application

Cypress does not allow you to change superdomains within a single test.

You may need to restructure some of your test code to avoid this problem.

Alternatively you can also disable Chrome Web Security which will turn off this restriction by setting { chromeWebSecurity: false } in your 'cypress.json' file.
```

**Comment by arun290636** [comment]
@jennifer-shehane Please try running this code few times- https://github.com/arun290636/cypress-issue.git . Cypress has stopped intermittently with below error :
Environment:
OS- Windows 7
NPM version 5.6
Cypress version 3.1.0

{ Error: unexpected end of file
    at Gunzip.zlibOnError (zlib.js:152:15)
 errno: -5, code: 'Z_BUF_ERROR' }
Error: unexpected end of file
    at Gunzip.zlibOnError (zlib.js:152:15)

As i said earlier this issue is a blocker for me .

**Comment by jennifer-shehane** [comment]
K, I was able to reproduce. It does take running it a few times. This is a definite bug. Thanks for giving us a reproducible example. 

```
  cypress:server:timers clearing timer id 1757 from queue { '4': { args: [], ms: 2500, cb: [Function: cb] }, '1523': { args: [], ms: 85000, cb: [Function] }, '1755': { args: [], ms: 85000, cb: [Function] }, '1756': { args: [], ms: 30000, cb: [Function] }, '1757': { args: [], ms: 3000, cb: [Function: timeoutTimeout] } } +1ms
{ Error: unexpected end of file
    at Gunzip.zlibOnError (zlib.js:152:15)
 errno: -5, code: 'Z_BUF_ERROR' }
Error: unexpected end of file
    at Gunzip.zlibOnError (zlib.js:152:15)

  cypress:server:browsers killing browser process +2ms
GET /fr/skin/frontend/smartbox_rwd_v1/default/img/payment-reassurance/credit-mutuel.png - - ms - -
GET /fr/js/salesforce/liveagent.js - - ms - -
$ 
```

**Comment by arun290636** [comment]
@brian-mann @jennifer-shehane Please let me know once this is fixed, As i said earlier this is blocker for us.

Thanks

**Comment by jennifer-shehane** [comment]
@arun290636 We are a small team and have many considerations on what issues we work on and when, and will get to this issue as soon as we can. We can prioritize bug fixes, do screensharing, and code reviews under our premium support. If you’re interested in this kind of premium support, please let us know by emailing support@cypress.io.

**Comment by arun290636** [comment]
@jennifer-shehane  Thanks for you quick response. I really liked Cypress as a tool but will stop using for now. Will wait for you guys to fix this issue. I do understand that you are a small team. Take your time, please let me know once this issue is fixed.

**Comment by jennifer-shehane** [comment]
I should have been more specific about what I did to replicate this bug. Step to reproduce:
1. Run the tests below (like in a Cy sandbox project) - I am on v3.0.1
2. Run `cypress open` 
3. Click on the test file with the test code
4. Run test code + refresh the page after each run until the error prints to terminal.

```js
it('add card details', function () {
  cy.visit('http://www.smartbox.com/fr/nos-smartbox/sejour/v794-sejour-gourmand-en-savoie.html')
  cy.get('.gdpr-modal-button-continue').click()
  cy.get('span.add-to-cart__content').first().click()
  cy.get('a.button.primary.no-margin-bottom').click()
})
```

**Comment by brian-mann** [comment]
I have discovered the root underlying cause of this and am working on a fix.

**Comment by arun290636** [comment]
@brian-mann when are you planning to release 3.0.2 ?

**Comment by brian-mann** [comment]
Released in [`3.0.2`](https://docs.cypress.io/guides/references/changelog.html#3-0-2).
